### ISSUE TITLE
Finding the best parameters for trining

### ISSUE THREAD

**Issue by niddal-imam** [Holmeyoung/crnn-pytorch#8]
I have 304 training sample and 60 val samples,. I have been trying to train the model, but the accuracy started with 0.008333 and stays 0.000000. What would be the best parameters for my data?
![image](https://user-images.githubusercontent.com/47844140/57290575-b6bb3d00-708b-11e9-9ebe-e70700582333.png)

**Comment by Holmeyoung** [comment]
Hi, i think the problem is that your training sample is too little. The proper number should be `number = 1000*len(alphabet)`. Try to make more data and you will get good result. As for the best parameters, just keep the default
```python
# finetune
nepoch = 1000 # number of epochs to train for
batchSize = 64 # input batch size
lr = 0.001 # learning rate for Critic, not used by adadealta
beta1 = 0.5 # beta1 for adam. default=0.5
adam = False # whether to use adam (default is rmsprop)
adadelta = False # whether to use adadelta (default is rmsprop)
```

**Comment by niddal-imam** [comment]
Thank you very much.

**Comment by mariembenslama** [comment]
P.S: I noticed that when an accuracy doesn't add up, it may be because the text length is variable. 
So you can train the model on a fixed length (10 for example) but when giving it a new image (even the the length is not 10), it'q able to recognize it.

Right @Holmeyoung?

**Comment by Holmeyoung** [comment]
@mariembenslama When an accuracy doesn't add up, it's more likely that the the data is not enough. As for training the model on a fixed length and prediting for different, 
```python
class resizeNormalize(object):

    def __init__(self, size, interpolation=Image.BILINEAR):
        self.size = size
        self.interpolation = interpolation
        self.toTensor = transforms.ToTensor()

    def __call__(self, img):
        img = img.resize(self.size, self.interpolation)
        img = self.toTensor(img)
        img.sub_(0.5).div_(0.5)
        return img
```
We can see the image is resized to `[32, 100]`, so if no matter your image label is 6 or 16 words, your image length is all `x*y`, you can train in fixes lengh, or you can't. Becaues after resized, the words in `32*100` image and `32*200` image are in different shape.

**Comment by niddal-imam** [comment]
Thank you both for your suggestions. I have collected more data (1000 images) and tried to train the model, but accuracy still does not add up. Perhaps because as you have mentioned before that the number = 1000*len(alphabet). I have 30 alphabet characters, so I need 30000 training sample. I tried to use a text recognition data generator, but unfortunately it can not generate Arabic text properly. However, I will try to find more training samples.
